src/server/transaction.h

```diff
@@ -500,6 +498,7 @@ class Transaction {
   // Multi transactions unlock asynchronously, so they need to keep a copy of all they keys.
   // "Launder" keys by filtering uniques and replacing pointers with same lifetime as transaction.
   void LaunderKeyStorage(CmdArgVec* keys);
+  bool CheckLock(EngineShard* shard, IntentLock::Mode mode, const KeyLockArgs& lock_args);
```

Comment:
1. Please add a comment
2. Why did you move it here? Why do Release and Acquire work with KeyLockArgs and this one is separate? More exceptions - > more diffciult code
3. Confusing signature - this function is a member, but doesn't access anything

Reply:
Moreover, KeyLockArgs if free to construct for, why bother with separation

Reply:
I removed the function in DbSlice. I removed the code,  not added it.

Reply:
You _moved_ it here 🙂 More specifically, to be a member. I'm asking why

Reply:
removed